gstd/src/msg/encoded.rs

```diff
@@ -113,6 +125,34 @@ pub fn reply<E: Encode>(payload: E, value: u128) -> Result<MessageId> {
     super::reply_bytes(payload.encode(), value)
 }
 
+/// Same as [reply], but encodes payload to stack allocated buffer.
+/// Buffer size for encoding is at least `E::max_encoded_len()`.
+pub fn reply_on_stack<E: Encode + MaxEncodedLen>(payload: E, value: u128) -> Result<MessageId> {
```

Comment:
Can't we avoid different methods and append this thing into already existing "reply" that require `Encode` because `encoded_size` must work without extra allocations https://github.com/paritytech/parity-scale-codec/blob/07c7e2c5a34da36684339710d46e2ea02a478dd6/src/codec.rs#L257

Reply:
just branch logic depending on encoding size

Reply:
> The problem is that each type which cannot be represented as MaxEncodedLen requires allocations. So, encoded_size must do additional allocation. We can skip this in future by using special self allocator for self output object.

But docs state other thing :/

Reply:
> However, this can not prevent allocations that some types are doing inside their own encoding.

It may be about custom Encode implementations
Need to check defaults

Reply:
Ok, as I can see this comment is wrong. Actually in most cases this really will be allocation free operation.

Reply:
I will add issue and TODO

Reply:
TODO added